src/redux/books/booksSlice.js

```diff
@@ -0,0 +1,31 @@
+import { createSlice } from '@reduxjs/toolkit';
+import { v4 as uuidv4 } from 'uuid';
+
+const initialState = {
+  bookList: [],
+};
+
+export const booksSlice = createSlice({
+  name: 'books',
+  initialState,
+  reducers: {
+    add: (state, action) => {
+      const newObject = {
+        title: action.payload.title,
+        author: action.payload.author,
+        id: uuidv4(),
+      };
+      state.bookList.push(newObject);
```

Comment:
- [ ] When creating a new book object, first you need to copy the previous books with `...state.books` and then add on that, because as it is currently the previous books will be deleted from the array and only the last added book will remain.
![image](https://user-images.githubusercontent.com/62705947/221688439-c206fb2e-0c28-4ccd-bbe5-8f2bb5ddc6cf.png)

Reply:
@GzimAsani  I checked my function many times and i added and deleted many books and this didn't happen.